projects/make-a-vis/src/lib/data-view/main/main.component.ts

```diff
@@ -9,13 +9,23 @@ import { DataService, DataSource } from '../shared/data.service';
   templateUrl: './main.component.html',
   styleUrls: ['./main.component.scss']
 })
-export class MainComponent implements OnInit {
+export class MainComponent implements OnDestroy {
+  /**
+   * Data sources observable
+   */
   dataSources: Observable<DataSource[]>;
+  /**
+   * Data subscription for the data-sources
+   */
+  dataSubscription: Subscription;
+  /**
+   * Boolean to show initial info message in the data view
+   */
   showDataViewMessage = true;
 
   constructor(private dataService: DataService) {
-    this.dataSources = dataService.dataSourcesChanged;
-    this.dataSources.subscribe((d) => {
+    this.dataSources = this.dataService.dataSourcesChanged;
```

Comment:
No good reason to add `this.` to dataService.dataSourcesChanged.

Reply:
VSCode shows the dataservice instance in constructor arguments as greyed out without `this`, says value is declared but never read.

Reply:
Oh I see, you can remove the private keyword to make it a local (to the constructor) variable. That should stop it from complaining.